### PR TITLE
Add filters to show Draft salary slips in salary register

### PR DESCRIPTION
**Pull request by PawanMeh**
Issue: #14778 

Testing video

salary_register

### CODE REVIEW

File: erpnext/hr/report/salary_register/salary_register.py [hunk 1]
```diff
@@ -74,13 +74,23 @@ def get_columns(salary_slips):
...
-	salary_slips = frappe.db.sql("""select * from `tabSalary Slip` where docstatus = 1 %s
+	salary_slips = frappe.db.sql("""select * from `tabSalary Slip` where %s
 		order by employee""" % conditions, filters, as_dict=1)
 
 	return salary_slips or []
 
 def get_conditions(filters):
 	conditions = ""
+
+	if filters.get("docstatus"):
```
Comment (Zlash65) [inline erpnext/hr/report/salary_register/salary_register.py]: Maybe we can skip the if-elif-elif block ?
```
doc_status = {"Draft": 0, "Submitted": 1, "Cancelled": 2}

if filters.get("docstatus"):
	conditions += "docstatus = {0}".format(doc_status(filter.get("docstatus")))
```

Comment (PawanMeh) [reply erpnext/hr/report/salary_register/salary_register.py]: Tried to use the suggested code and got the below error. 

conditions += "docstatus = {0}".format(doc_status(filter.get("docstatus")))
AttributeError: 'builtin_function_or_method' object has no attribute 'get'

I think we will have to loop through the dict after getting the filter value and then use the value in format? Is that better than if-elif-elif?

Comment (Zlash65) [reply erpnext/hr/report/salary_register/salary_register.py]: My bad, it should be filters.get("docstatus") - missed an s there

Comment (PawanMeh) [reply erpnext/hr/report/salary_register/salary_register.py]: conditions += "docstatus = {0}".format(doc_status(filters.get("docstatus")))
TypeError: 'dict' object is not callable

**Pull request merged**